apps/dashboard/services/graphql/queries/get-transactions.ts

```diff
@@ -0,0 +1,200 @@
+                ... on InitiationViaIntraLedger {
+                  counterPartyUsername
+                  counterPartyWalletId
+                }
+                ... on InitiationViaOnChain {
+                  address
+                }
+                ... on InitiationViaLn {
+                  paymentHash
+                }
+              }
+            }
+          }
+          pageInfo {
+            endCursor
+            hasNextPage
+            hasPreviousPage
+            startCursor
+          }
+        }
+      }
+    }
+  }
+`;
+
+gql`
+  query GetFirstTransactions($first: Int) {
+    me {
+      id
+      defaultAccount {
+        transactions(first: $first) {
+          edges {
+            cursor
+            node {
+              createdAt
+              direction
+              id
+              memo
+              settlementAmount
+              settlementCurrency
+              settlementDisplayAmount
+              settlementDisplayCurrency
+              settlementDisplayFee
+              settlementFee
+              settlementVia {
+                ... on SettlementViaIntraLedger {
+                  counterPartyUsername
+                  counterPartyWalletId
+                }
+                ... on SettlementViaLn {
+                  paymentSecret
+                  preImage
+                }
+                ... on SettlementViaOnChain {
+                  transactionHash
+                  vout
+                }
+              }
+              status
+              settlementPrice {
+                base
+                currencyUnit
+                formattedAmount
+                offset
+              }
+              initiationVia {
+                ... on InitiationViaIntraLedger {
+                  counterPartyUsername
+                  counterPartyWalletId
+                }
+                ... on InitiationViaOnChain {
+                  address
+                }
+                ... on InitiationViaLn {
+                  paymentHash
+                }
+              }
+            }
+          }
+          pageInfo {
+            endCursor
+            hasNextPage
+            hasPreviousPage
+            startCursor
+          }
+        }
+      }
+    }
+  }
+`;
+
+export async function fetchFirstTransactions(
+  token: string,
+  first: number = 10
+) {
+  const client = apollo(token).getClient();
+
+  try {
+    const data = await client.query<GetFirstTransactionsQuery>({
+      query: GetFirstTransactionsDocument,
+      variables: { first },
+    });
+    return data.data.me?.defaultAccount.transactions;
+  } catch (err) {
+    console.error("error in getting Transaction details", err);
+    if (err instanceof Error) {
+      throw new Error(err.message);
+    }
+    throw new Error("Unknown error");
+  }
+}
+
+export async function fetchPaginatedTransactions(
+  token: string,
+  direction: "next" | "previous",
+  cursor: string | null = null,
+  first: number = 10
```

Comment:
we could show the first 50 tx instead of first 10. it will make it less likely to have a need to actually use the pagination service (specially useful if there is a bug in the previous for pagination)